cwl/src/main/scala/cwl/CommandOutputBinding.scala

```diff
@@ -1,15 +1,212 @@
 package cwl
 
-import cwl.CommandOutputBinding.Glob
-import shapeless.{:+:, CNil}
+import cats.instances.list._
+import cats.syntax.apply._
+import cats.syntax.traverse._
+import cats.syntax.validated._
+import common.validation.ErrorOr._
+import common.validation.Validation._
+import wom.expression.IoFunctionSet
+import wom.types._
+import wom.values._
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
 
 /** @see <a href="http://www.commonwl.org/v1.0/Workflow.html#CommandOutputBinding">CommandOutputBinding</a> */
-case class CommandOutputBinding(
-                                 glob: Option[Glob] = None,
-                                 loadContents: Option[Boolean] = None,
-                                 outputEval: Option[StringOrExpression] = None)
+case class CommandOutputBinding(glob: Option[Glob] = None,
+                                loadContents: Option[Boolean] = None,
+                                outputEval: Option[StringOrExpression] = None)
 
 object CommandOutputBinding {
-  type Glob = Expression :+: String :+: Array[String] :+: CNil
   val ReadLimit = Option(64 * 1024)
+
+  /**
+    * Returns all the primary and secondary files that _will be_ created by this command output binding.
+    */
+  def getOutputWomFiles(inputValues: Map[String, WomValue],
```

Comment:
Does it make sense to include this in `CommandOutputExpression`  as it is doing expression evaluation?  Or perhaps creating a secondaryfilesExpression class?  Would be nice to separate this from the model IMO

Reply:
Happy if all this business logic lives in some other location. It's all static code that can live under any `object` now.